.github/workflows/canister-tests.yml

```diff
@@ -289,12 +289,19 @@ jobs:
           echo "The following screenshots were created:"
           shasum -a 256 screenshots/*.png | sort -k2 # sort by 2nd column (filename)
 
-      - name: Archive screenshots
-        if: ${{ always() }}
+      - name: Upload screenshots
         uses: actions/upload-artifact@v2
         with:
           name: e2e-screenshots-${{ matrix.device }} ${{ matrix.start-flag }}
-          path: screenshots/**/*.png
+          path: screenshots/*.png
+
+      - name: Archive test failures
+        if: ${{ always() }}
```

Comment:
why is this needed and what does it do?

Reply:
So I have changed the file structure of the selenium errors:
It previously used to be `screenshots/<name>.png` for normal screenshots and `screenshots/error/<time>.png` for errors. Now its `screenshots/<name>.png` for normal screenshots. And a new artifact that collects a screenshot, the page source of the page that failed and the browser logs at that moment in `test-failures/<test_name>.png|html|log`.

And this is why I had to touch the github actions workflow.

Reply:
but why the `if: always()`? This looks like `if (true)` when I read it. That's why I'm asking.

Reply:
Ah, normally github actions just aborts the workflow if it fails. But for test-failure artifacts we absolutely want the files anyway. So it's `always` as opposed to the default which is only on success.

Reply:
aaahhh, ok, got it. Thanks!